cmd/nerdctl/run.go

```diff
@@ -488,11 +489,20 @@ func createContainer(cmd *cobra.Command, ctx context.Context, client *containerd
 		if err != nil {
 			return nil, "", nil, err
 		}
-		switch logDriver {
-		case "json-file", "journald", "fluentd":
-		default:
+		if !strutil.InStringSlice(logging.Drivers(), logDriver) {
 			return nil, "", nil, fmt.Errorf("unknown driver %q", logDriver)
 		}
+		switch logDriver {
+		case "json-file":
+			// Initialize the log file (https://github.com/containerd/nerdctl/issues/1071)
+			// TODO: move this logic to pkg/logging
+			jsonFilePath := jsonfile.Path(dataStore, ns, id)
+			if _, err := os.Stat(jsonFilePath); errors.Is(err, os.ErrNotExist) {
+				if writeErr := os.WriteFile(jsonFilePath, []byte{}, 0600); writeErr != nil {
+					return nil, "", nil, writeErr
+				}
+			}
+		}
```

Comment:
I think a better place for creating the json log file would be here https://github.com/containerd/nerdctl/blob/a7fdf67869d00b3ffeaa86dd70237bad9a228a25/pkg/logging/json_logger.go#L36 
You mention it as TODO why not doing it now ?

Reply:
Sorry I take back what I said :) , SGTM

Reply:
fixed